$Id: fix_rssreader.patch 419 2012-12-29 22:39:54Z tg $

Fix RSS_Reader extension:
* apply lastRSS patch as shown on upstream website (Debian-originated)
* set $egCache to false by default, since we have no www-data writable
  cache directory, and I don't think we want it

Also add documentation of these changes as README.Debian and point to
upstream's documentation in form of a wikipage.

--- a/dist/mediawiki-extensions-base/usr/share/mediawiki-extensions/base/RSS_Reader/RSSReader.php
+++ b/dist/mediawiki-extensions-base/usr/share/mediawiki-extensions/base/RSS_Reader/RSSReader.php
@@ -39,7 +39,7 @@ $egRSSReaderPath  = $wgScriptPath."/exte
 $egCacheTime      = 3600; //default cache time in seconds
 $egCacheTimeMin   = 1800; //minimum cache time in seconds
 $egCacheTimeMax   = 7200; //maximum cache time in seconds
-$egCache          = true; //boolean to determine if caching should be done
+$egCache          = false; //boolean to determine if caching should be done
 $egCacheDir       = dirname( __FILE__ ).'/cache/'; //directory of cache
 //boolean to determine if links created should have rel="nofollow"
 $egNoFollow       = false;
--- a/dist/mediawiki-extensions-base/usr/share/mediawiki-extensions/base/RSS_Reader/lastRSS.php
+++ b/dist/mediawiki-extensions-base/usr/share/mediawiki-extensions/base/RSS_Reader/lastRSS.php
@@ -149,14 +149,14 @@ class lastRSS {
 				{ $this->rsscp = $this->default_cp; } // This is used in my_preg_match()
 
 			// Parse CHANNEL info
-			preg_match("'<channel.*?>(.*?)</channel>'si", $rss_content, $out_channel);
+			preg_match("'<channel.*?>(.*)</channel>'si", $rss_content, $out_channel);
 			foreach($this->channeltags as $channeltag)
 			{
 				$temp = $this->my_preg_match("'<$channeltag.*?>(.*?)</$channeltag>'si", $out_channel[1]);
 				if ($temp != '') $result[$channeltag] = $temp; // Set only if not empty
 			}
 			// If date_format is specified and lastBuildDate is valid
-			if ($this->date_format != '' && ($timestamp = strtotime($result['lastBuildDate'])) !==-1) {
+			if ($this->date_format != '' && isset($result['lastBuildDate']) && ($timestamp = strtotime($result['lastBuildDate'])) !==-1) {
 						// convert lastBuildDate to specified date format
 						$result['lastBuildDate'] = date($this->date_format, $timestamp);
 			}
@@ -198,7 +198,7 @@ class lastRSS {
 					if ($this->stripHTML && $result['items'][$i]['title'])
 						$result['items'][$i]['title'] = strip_tags($this->unhtmlentities(strip_tags($result['items'][$i]['title'])));
 					// If date_format is specified and pubDate is valid
-					if ($this->date_format != '' && ($timestamp = strtotime($result['items'][$i]['pubDate'])) !==-1) {
+					if ($this->date_format != '' && isset($result['items'][$i]['pubDate']) && ($timestamp = strtotime($result['items'][$i]['pubDate'])) !== -1) {
 						// convert pubDate to specified date format
 						$result['items'][$i]['pubDate'] = date($this->date_format, $timestamp);
 					}
--- /dev/null
+++ b/dist/mediawiki-extensions-base/usr/share/doc/mediawiki-extensions/base/RSS_Reader/README.Debian
@@ -0,0 +1,9 @@
+User-visible changes to the RSS_Reader extension:
+* set $egCache to false by default, since we have no www-data writable
+  cache directory, and I don't think we want it
+
+Upstream provided installation instruction for this version can be found at:
+http://www.mediawiki.org/w/index.php?title=Extension:RSS_Reader&oldid=622104
+
+
+$Id: fix_rssreader.patch 419 2012-12-29 22:39:54Z tg $
